RichClipboard-Tests.package/RCEditorGlobalStateHandlerTest.class/instance/testEditorSetsGlobalStateOnAltAndCtrl.st

```diff
@@ -0,0 +1,54 @@
+testing
+testEditorSetsGlobalStateOnAltAndCtrl
+	| priorAlt priorCtrl textMorph |
+	
+	priorAlt := RCEditorGlobalStateHandler altKeyDown.
+	priorCtrl := RCEditorGlobalStateHandler ctrlKeyDown.
+	
+	RCEditorGlobalStateHandler altKeyDown: false.
+	RCEditorGlobalStateHandler ctrlKeyDown: false.
+	textMorph := TextMorph new.
+	
+	textMorph keyDown: (KeyboardEvent new
+			setType: #keyDown
+			buttons: 0
+			position: textMorph center
+			keyValue: 17
+			hand: nil
+			stamp: nil).
+	self assert: RCEditorGlobalStateHandler altKeyDown not.
+	self assert: RCEditorGlobalStateHandler ctrlKeyDown.
+	
+	textMorph keyDown: (KeyboardEvent new
+			setType: #keyDown
+			buttons: 0
+			position: textMorph center
+			keyValue: 18
+			hand: nil
+			stamp: nil).
+	self assert: RCEditorGlobalStateHandler altKeyDown.
+	self assert: RCEditorGlobalStateHandler ctrlKeyDown.
+	
+	textMorph keyUp: (KeyboardEvent new
+			setType: #keyUp
+			buttons: 0
+			position: textMorph center
+			keyValue: 17
+			hand: nil
+			stamp: nil).
+	self assert: RCEditorGlobalStateHandler altKeyDown.
+	self assert: RCEditorGlobalStateHandler ctrlKeyDown not.
+	
+	
+	textMorph keyUp: (KeyboardEvent new
+			setType: #keyUp
+			buttons: 0
+			position: textMorph center
+			keyValue: 18
+			hand: nil
+			stamp: nil).
+	self assert: RCEditorGlobalStateHandler altKeyDown not.
+	self assert: RCEditorGlobalStateHandler ctrlKeyDown not.
+	
+	RCEditorGlobalStateHandler altKeyDown: priorAlt.
+	RCEditorGlobalStateHandler ctrlKeyDown: priorCtrl
```

Comment:
Dieser Test ist meiner Meinung nach zu lang. Ich würde vorschlagen wir machen daraus zwei Tests. Einen Test für Alt und einen für Ctrl.

Reply:
> 
> 
> Warum haben wir keinen Test, der prüft, ob sich ein RCContextMenu öffnet?
> Wahrscheinlich habt ihr einen guten Grund, ich habe es aber nicht ganz verstanden

Weil das Kontextmenü leider mit der Live-Version von RichClipboard interagiert (wir also das "Isolated Test" Pattern verletzen würden). Ohne Refactoring vom Kontextmenü ist das also nicht so ohne weiteres möglich.

Das war schon beim RCEditorGlobalStateHandler problematisch, aber hier ist es relativ leicht, den alten Zustand wiederherzustellen und außerdem bermerkt der Benutzer es wahrscheinlich sowieso nicht, wenn hier was schiefgeht.

Reply:
> 
> 
> Dieser Test ist meiner Meinung nach zu lang. Ich würde vorschlagen wir machen daraus zwei Tests. Einen Test für Alt und einen für Ctrl.

Der Gedanke hinter diesem Test war es grade, die Interaktion von `alt` und `ctrl` zu testen. Was wir tun können, um den Test zu verkürzen, ist die KeyboardEvents in anderen Methoden zu erstellen und nur noch sowas wie `self altDownEvent` aufzurufen.

Eine andere Möglichkeit ist, das Speichern und Restoren des GlobalStateHandlers in setup und tearDown zu schieben.

Leo und ich hatten uns aber gegen beides entschieden, weil das unserer Meinung nur die Komplexität der Klasse unnötig erhöht hätte. Schließlich liegt in der ja nur ein einziger Test.